solr/core/src/test/org/apache/solr/search/neural/KnnQParserTest.java

```diff
@@ -39,7 +39,7 @@ public class KnnQParserTest extends SolrTestCaseJ4 {
   @Before
   public void prepareIndex() throws Exception {
     /* vectorDimension="4" similarityFunction="cosine" */
-    initCore("solrconfig-basic.xml", "schema-densevector.xml");
+    initCore("solrconfig_codec.xml", "schema-densevector.xml");
```

Comment:
1/3 I've been looking at the `DenseVectorFieldTest` test and also arrived at the conclusion that something other than `solrconfig-basic.xml` is needed there so that `SchemaCodecFactory` is used ...